### PR TITLE
term-entry: kotlin periodUntil

### LINKED ISSUE #3817: [Term Entry] Kotlin Datetime .periodUntil()
### Reason/inspiration (optional)

We would like a new term entry on the `.periodUntil()` method for Kotlin: [Datetime](https://www.codecademy.com/resources/docs/kotlin/datetime). The entry should go in a new file under `docs/content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by vrun1208**
### Description

closes #3817 
Added a new term entry for kotlin dateTime .periodUntil() method

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Date'
+  - 'Methods'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **periodUntil()** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months and days.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:15]:
```suggestion
The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months, and days.
```

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Example
+
+```kotlin
+val startDate = LocalDate.of(2023,11,2)
+val endDate = LocalDate.of(2023,11,3)
+
+val period = startDate.periodUntil(endDate)
+
+println("difference: $period.days day")
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:33]: The code is not compiling properly, please check if the code is working or not.

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+The **periodUntil()** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months and days.
+
+## Syntax
+
+```pseudo
+expect fun LocalDate.periodUntil(other: LocalDate): DatePeriod
+```
+
+`DatePeriod`gives the difference between two instants, decomposed into date and time components where all time components are equals to zero.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:23]: Use bullet points to explain the parameters and explain the LocalDate parameter too.

Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:23]:
```suggestion
`DatePeriod` gives the difference between two instants, decomposed into date and time components where all time components are equal to zero.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@vrun1208 Added some comments. Add a Note explaining if this function is replaced with some other function or is deprecated for some versions of Java.

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md (hunk 2 above)
Comment (vrun1208) [reply content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:33]: Hi @avdhoottt, where to add the prerequisites section? Inside the example or have to make a new section at the end?

Comment (avdhoottt) [reply content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:33]: Use this [guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) and write a note describing what the user will need in order to run this code. Add the note above the code.

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  val endDate = LocalDate.of(2023,11,3)
+
+  val period = startDate.periodUntil(endDate)
+
+  println("difference: $period.days day")
+}
+```
+
+`startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.
+
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:61]:
```suggestion

`startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.

The result of the above code is as follows:

```

**Review by avdhoottt — APPROVED** [review]
Good to go for the second review.

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Date'
+  - 'Methods'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months, and days.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:15]:
```suggestion
The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an object representing the difference between the two dates in terms of years, months, and days.
```

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+## Syntax
+
+```pseudo
+expect fun LocalDate.periodUntil(other: LocalDate): DatePeriod
+```
+
+- `DatePeriod` gives the difference between two instants, decomposed into date and time components where all time components are equal to zero.
+- `LocalDate` represents a specific civil date without a reference to a particular time zone.
+
+> **Note:** This method is fall under `kotlinx-datetime` library which is published to `Maven central`. Maven central repository should be added to the file first.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:26]:
```suggestion
> **Note:** This method is part of `kotlinx-datetime` library which is published to Maven central. Maven central repository should be added to the file first.
```

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+Maven central repository must be added to the `build.gradle.kts` file first if it is not already there,
```
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:28]:
```suggestion
Maven central repository must be added to the `build.gradle.kts` file first if it is not already there;
```

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  val startDate = LocalDate.of(2023,11,2)
+  val endDate = LocalDate.of(2023,11,3)
+
+  val period = startDate.periodUntil(endDate)
+
+  println("difference: $period.days day")
+}
+```
+
+`startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:60]: Can we rewrite so line 60 is before the code block example, introducing it?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@vrun1208 I did a second review and left a few comments.

File: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md (hunk 8 above)
Comment (vrun1208) [reply content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:60]: Hi @SSwiniarski, do you mean to move line 60 above ??

Comment (SSwiniarski) [reply content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md:60]: @vrun1208 above line 46 between `## Example` and the following code block.

**Review by SSwiniarski — APPROVED** [review]
@vrun1208 Looks good.

**Comment by caupolicandiaz** [comment]
@vrun1208 I've conducted an additional review and made some direct changes to clarify some of the content.

**Review by SSwiniarski — APPROVED** [review]

**Pull request merged**